Sources/RTMP/RTMPConnection.swift

```diff
@@ -282,11 +284,12 @@ public class RTMPConnection: EventDispatcher {
     }
 
     /// Creates a two-way connection to an application on RTMP Server.
-    public func connect(_ command: String, arguments: Any?...) {
+    public func connect(_ command: String, name: String? = nil, arguments: Any?...) {
```

Comment:
[MUST]
I don't want to make such disruptive changes for the purpose of supporting releaseStream. Also, considering viewing or multiple streams, I still think this change is unnecessary.

If I was to do it, I think it would be better to prepare an RTMPConnection subclass or wrapper class, such as RTMPConnectionFMLECompatible, that conforms to the FMLE sequence.

Reply:
Should I make RTMPConnection and `createStream()` open to allow subclassing? My goal is to have any kind of solution for this case being supported. It's common and solves platform specific issues while not causing others because all other popular encoders perform it in the mentioned sequence. FMLE app performs it in the following order:

> AMF command 'releaseStream'
> AMF command 'FCPublish'
> AMF command 'createStream'

Reply:
Yes. HaishinKit aims to provide multi-streaming and ingest/playback functionalities. Killing off RTMPConnection functionality or making destructive changes to methods in pursuit of FMLE compatibility would diverge from this objective.

Therefore, I believe it would be appropriate to have a class dedicated to single-stream ingest, inheriting from RTMPConnection, and within that class, call the FMLE-compatible methods.

Reply:
It seems natural for the stream name of releaseStream to be in RTMPStream.
To avoid issues with multi-streaming and playback without inheriting RTMPConnection, this implementation seems necessary.
https://github.com/shogo4405/HaishinKit.swift/pull/1393/files

It might be good to handle errors when there is a contradiction between the publish and fcPublishName, but I don't think it's necessary to be that strict.